rust/README.md

```diff
@@ -182,15 +182,20 @@ all:
 
 ## Mount caches and parallelization
 
-This library uses several mount caches per tuple of `{project, os_release}`:
+As of today, local Cargo builds cannot run in parallel for a given project, given that the output artifact folder is globally locked for the whole build.
+This library overcomes such limitation by using different mount caches for the target folder, one per Earthly target.
+While multiple concurrent builds of the same Earthly target would still block, the user now has the choice of creating new Earthly targets instead of reusing the same to increase parallelization.  
+
+Hence, this library uses several mount caches per tuple of `{project, os_release}`:
 - One cache mount for `$CARGO_HOME`, shared across all target builds without any locking involved. 
-- A family of locked cache mounts for `$CARGO_TARGET_DIR`. One per target. 
+- A family of locked cache mounts for `$CARGO_TARGET_DIR`. One per Earthly target. 
 
 Notice that:
-- the previous targets builds might belong to one or multiple Earthly builds.
+- the previous targets builds might belong to one or multiple Earthly builds, given that the caches involved are global
 - builds will only be blocked by concurrent ones of the same target
+- Earthly target builds are atomic in the sense that the artifacts returned are guaranteed to be the ones generated by that build.
 
 For example, running `earthly +all` in the previous example will:
 - run all targets (`+lint,+build,+test,+fmt,+check-dependencies`) in parallel without any blocking involved
 - use a common cache mount for `$CARGO_HOME`
-- use one individual `$CARGO_TARGET_DIR` cache mount per target 
+- use one individual `$CARGO_TARGET_DIR` cache mount per each target 
```

Comment:
I think `per target` was correct.

`for each target` would be an valid alternative, but `per each target` reads a bit odd to me.

Maybe you also want to inject an `Earthly`?

Reply:
thanks Ian, sure